src/containers/PaymentMethodsPage/PaymentMethodsPage.js

```diff
@@ -19,11 +26,90 @@ import {
 } from '../../components';
 import { TopbarContainer } from '..';
 import { PaymentMethodsForm } from '../../forms';
+import { createStripeSetupIntent, stripeCustomer, loadData } from './PaymentMethodsPage.duck.js';
 
 import css from './PaymentMethodsPage.css';
 
-export const PaymentMethodsPageComponent = props => {
-  const { currentUser, scrollingDisabled, intl } = props;
+const PaymentMethodsPageComponent = props => {
+  const {
+    currentUser,
+    addPaymentMethodError,
+    deletePaymentMethodError,
+    createStripeCustomerError,
+    handleCardSetupError,
+    onCreateSetupIntent,
+    onHandleCardSetup,
+    onCreateStripeCustomer,
+    onAddPaymentMethod,
+    onUpdatePaymentMethod,
+    onDeletePaymentMethod,
+    fetchStripeCustomer,
+    scrollingDisabled,
+    intl,
+  } = props;
+
+  const handleSubmit = params => {
+    const ensuredCurrentUser = ensureCurrentUser(currentUser);
+    const stripeCustomer = ensuredCurrentUser.stripeCustomer;
+    const hasSavedPaymentMethod = stripeCustomer ? stripeCustomer.defaultPaymentMethod : null;
+
+    const { stripe, card, formValues } = params;
+
+    onCreateSetupIntent().then(setupIntent => {
```

Comment:
If I read this correctly, you first create `onCreateSetupIntent` and then if it succeeds, you call `onHandleCardSetup `, but there's no error handling/catch block.

How about a bit more readable chaining:

```js
onCreateSetupIntent()
  .then(setupIntent => {
    const stripeParams = { 
      stripe, 
      card, 
      setupIntentClientSecret: getClientSecret(setupIntent), // new function to extract clientSecret
      paymentParams: getPaymentParams(currentUser, formValues), // new function to extract paymentParams
    };

    return onHandleCardSetup(stripeParams); // the return value will be passed as params to the next `then` block.
  })
  .then(result => {
    const newPaymentMethod = result.setupIntent.payment_method;
    // Note: stripe.handleCardSetup returns erros as successful call (200), but those are rejected in thunk functions.

    return !stripeCustomer
      ? onCreateStripeCustomer(newPaymentMethod)
      : hasSavedPaymentMethod
      ? onUpdatePaymentMethod(newPaymentMethod)
      : onAddPaymentMethod(newPaymentMethod);
  })
  .then(() => {
    // Update currentUser entity and its sub entities: stripeCustomer and defaultPaymentMethod
    fetchStripeCustomer();
  });
  .catch(error => {
    // Handle all possible errors from onCreateSetupIntent and onHandleCardSetup if needed.
    // If there's no need, it's still good practice to always add catch and then explain in the comments why it's empty.
  });
  
```